src/ck-core/ckfutures.C

```diff
@@ -262,6 +267,36 @@ void *CkWaitFutureID(CkFutureID handle)
   return value;
 }
 
+std::pair<void*, CkFutureID> CkWaitAnyID(const std::vector<CkFutureID>& handles) {
+  auto fs = &(CpvAccess(futurestate));
+  const auto ready = std::any_of(handles.begin(), handles.end(),
+    [&fs](const CkFutureID& id) { return fs->is_ready(id); });
+  /* A single request is generated, and enqueued for all the futures that we are
+   * interested in. Note, a request may only be fulfilled once, then it will expire
+   * and become a no-op; thus, the corresponding `CthAwaken` for this thread will
+   * only be called once (for more details, see FutureToThread::fulfill).
+   */ 
+  if (!ready) fs->request(handles, std::make_shared<FutureToThread>(CthSelf()));
+  do {
+    if (!ready) CthSuspend();
+    for (const auto& handle : handles) {
+      if (fs->is_ready(handle)) {
+        return std::make_pair((*fs)[handle], handle);
+      }
+    }
+  } while (true);
+}
+
+std::vector<void*> CkWaitAllIDs(const std::vector<CkFutureID>& ids) {
+  // a more sophisticated solution should be implemented that interleaves
```

Comment:
This seems worth doing now, and doesn't seem like it would be difficult to implement if I understand correctly. Separately, it would be better to initialize the vector to size and then assign rather than repeatedly calling push_back.

Reply:
Perhaps this comment should be deleted, or hedged somewhat, since the more sophisticated approach is unlikely to make a significant difference at this time. As `CkWaitFuture` need not be called for a value to arrive, both approaches are constrained by the wait time for the last future to arrive. The more sophisticated approach could, potentially, have fewer fewer invocations of CthSuspend/CthAwaken... but it's unclear to me whether that would be worthwhile or not, as the longest-wait-time is invariant and the dominant performance factor.